src/utils/api/fetch.ts

```diff
@@ -0,0 +1,77 @@
+export enum METHODS {
+  GET = "GET",
+  PUT = "PUT",
+  POST = "POST",
+  DELETE = "DELETE",
+}
+
+interface HTTPOptions {
+  timeout?: number;
+  headers?: Record<string, string>;
+  data?: Record<string, any> | null;
+}
+
+export class HTTPTransport {
+  baseUrl = "https://ya-praktikum.tech/api/v2/";
+
+  get = (url: string, options: HTTPOptions = {}): Promise<XMLHttpRequest> => {
```

Comment:
```suggestion
  get = <ResponseType,>(url: string, options: HTTPOptions = {}): Promise<ResponseType> => {
```

Наверное что-то подобное тут хочется видеть